src/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs

```diff
@@ -731,18 +736,33 @@ internal abstract partial class SafeDeleteContext : SafeHandle
 
                         if (NetEventSource.IsEnabled) NetEventSource.Info(null, "Marshaling OUT buffer");
 
-                        // Get unmanaged buffer with index 0 as the only one passed into PInvoke.
-                        outSecBuffer.size = outUnmanagedBuffer.cbBuffer;
-                        outSecBuffer.type = outUnmanagedBuffer.BufferType;
-                        outSecBuffer.token = outUnmanagedBuffer.cbBuffer > 0 ?
-                            new Span<byte>((byte*)outUnmanagedBuffer.pvBuffer, outUnmanagedBuffer.cbBuffer).ToArray() :
-                            null;
+                        // No data written out but there is Alert
+                        if (outUnmanagedBuffer[0].cbBuffer == 0 && outUnmanagedBuffer[1].cbBuffer > 0)
+                        {
+                            outSecBuffer.size = outUnmanagedBuffer[1].cbBuffer;
+                            outSecBuffer.type = outUnmanagedBuffer[1].BufferType;
+                            outSecBuffer.token = new Span<byte>((byte*)outUnmanagedBuffer[1].pvBuffer, outUnmanagedBuffer[1].cbBuffer).ToArray();
+                        }
+                        else
+                        {
+                             outSecBuffer.size = outUnmanagedBuffer[0].cbBuffer;
+                             outSecBuffer.type = outUnmanagedBuffer[0].BufferType;
+                             outSecBuffer.token = outUnmanagedBuffer[0].cbBuffer > 0 ?
+                                 new Span<byte>((byte*)outUnmanagedBuffer[0].pvBuffer, outUnmanagedBuffer[0].cbBuffer).ToArray() :
+                                 null;
+                        }
                     }
                 }
             }
             finally
             {
                 outFreeContextBuffer?.Dispose();
+                if (outUnmanagedBuffer[1].pvBuffer != IntPtr.Zero)
+                {
+                    outFreeContextBuffer = SafeFreeContextBuffer.CreateEmptyHandle();
+                    outFreeContextBuffer.Set(outUnmanagedBuffer[1].pvBuffer);
+                    outFreeContextBuffer.Dispose();
```

Comment:
Seems a little wasteful to allocate a new SafeHandle just for the purposes of using its ReleaseHandle method.  Can we instead just do:
```C#
if (outUnmanagedBuffer[1].pvBuffer != IntPtr.Zero)
{
    Interop.SspiCli.FreeContextBuffer(outUnmanagedBuffer[1].pvBuffer);
}
```
?

Reply:
I'll give it try. I followed existing pattern but for alert, this was never passed in and it will never leave this function.

Reply:
updated.